android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskExpireUtil.kt

```diff
@@ -58,28 +55,22 @@ constructor(@ApplicationContext val appContext: Context, val fhirEngine: FhirEng
             { value = of(Task.TaskStatus.INPROGRESS.toCoding()) },
             { value = of(Task.TaskStatus.RECEIVED.toCoding()) },
           )
-          if (lastAuthoredOnDate != null) {
-            filter(
-              Task.AUTHORED_ON,
-              {
-                prefix = ParamPrefixEnum.GREATERTHAN_OR_EQUALS
-                value = of(DateTimeType(lastAuthoredOnDate))
-              }
-            )
-          }
-          count = tasksCount
-          sort(Task.AUTHORED_ON, Order.ASCENDING)
```

Comment:
The authored on was used here to limit the amount of data to process at a time, otherwise a pre-configured batch size would be used. This background process is expensive and blocking especially during the database reads.